### PR TITLE
 waylandpp: 0.2.10 -> 1.0.0 

### PR DESCRIPTION
**Pull request by minijackson**
###### Description of changes

Changes: https://github.com/NilsBrause/waylandpp/releases/tag/1.0.0

Despite the major version, there are only ABI breaking changes in this release. Since the reason for the major change is only for adding Wayland Server support, I don't think it warrants a release note entry.

The only reverse-dependency that I know of is kodi-wayland (confirmed by nix-review) and it seems to work fine.

Also some minor changes to silence the Fontconfig warnings in the build logs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by minijackson** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kodi-wayland</li>
    <li>waylandpp</li>
  </ul>
</details>

**Comment by minijackson** [comment]
@ofborg build waylandpp

File: pkgs/development/libraries/waylandpp/default.nix [hunk 1]
```diff
@@ -17,13 +18,13 @@ assert docSupport -> doxygen != null;
 with lib;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waylandpp/default.nix:18]:
```suggestion
```
Please apply more locally where needed.

File: pkgs/development/libraries/waylandpp/default.nix [hunk 2]
```diff
@@ -32,6 +33,11 @@ stdenv.mkDerivation rec {
     "-DWAYLAND_SCANNERPP=${buildPackages.waylandpp}/bin/wayland-scanner++"
   ];
 
+  # Complains about not being able to find the fontconfig config file otherwise
+  FONTCONFIG_FILE = optional docSupport (makeFontsConf { fontDirectories = [ ]; });
+  # Resolves the warning "Fontconfig error: No writable cache directories"
+  XDG_CACHE_HOME = "/tmp";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waylandpp/default.nix:39]: Please use ``mktemp -d`` and use the output as that in a variable.

``` 
preBuild = ''
  export XDG_CACHE_HOME=$(mktemp -d)
'';
```

File: pkgs/development/libraries/waylandpp/default.nix [hunk 3]
```diff
@@ -14,16 +15,15 @@
 
 assert docSupport -> doxygen != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waylandpp/default.nix:17]:
```suggestion
```
Those are no longer required and an artifact from before callPackage. Also please drop ``? null`` from inputs.

**Pull request merged**